### PR TITLE
Docs/getting started

### PR DESCRIPTION
**Pull request by kmdeck**
# Description
Updates to the getting started pages within the docs in response to feedback.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: docs/src/GettingStarted/Terminology.md [hunk 1]
```diff
@@ -4,13 +4,13 @@ The `ClimateMachine` documentation uses terminology from several disciplines. Be
...
+* Driver file: a julia script which (a) specifies which [`BalanceLaw`](#ClimateMachine.BalanceLaws.BalanceLaw) is being used, (b) specifies changes to the default version of that model, (c) specifies initial and boundary conditions, (d) specifies numerical details related to timestep, grid, and frequency of output, and (e) runs the integration. Examples can be found in tutorials and experiments.
 * Source file: a julia file containing code involved in implementing the choices made in a driver file. This is where the numerical methods related to solving the equations (e.g. the ODE time steppers, or the creation of the grid) are defined. Users typically only interact with functions defined in source code within a driver file by using arguments to those functions, unless they wish to develop the source code itself.
 * Kernel: pre-compiled functions designed for efficiency on GPUs.
```
Comment (jkozdon) [inline docs/src/GettingStarted/Terminology.md:9]:
```suggestion
* Kernel: Functions which are launched on the compute device (i.e., the CPU or GPU) and run with an array of workitems or threads. `ClimateMachine.jl` uses [`KernelAbstractions.jl`](https://github.com/JuliaGPU/KernelAbstractions.jl) as its kernel language.
```

File: docs/src/GettingStarted/Terminology.md [hunk 2]
```diff
@@ -4,13 +4,13 @@ The `ClimateMachine` documentation uses terminology from several disciplines. Be
...
-
+* Callback:
```
Comment (jkozdon) [inline docs/src/GettingStarted/Terminology.md:10]:
```suggestion
* Callback: Functions executed by the ODE integrator after each time step; see
            [`solve!'`](#ClimateMachine.ODESolvers.solve!).
            This allows the ability to inject custom behavior into the ODE
            integrators, such as diagnostic output, visualization, and apply
            filtering to the numerical solution.
            Several convenience functions exists specifying callback frequency:
            [number of simulation steps](#ClimateMachine.GenericCallbacks.EveryXSimulationSteps),
            [every X simulation time](#ClimateMachine.GenericCallbacks.EveryXSimulationTime),
            and
            [every X wall clock seconds](#ClimateMachine.GenericCallbacks.EveryXWallTimeSeconds).
            The callback mechanism is inspired by the [callback mechanism of
            `DifferentialEquations.jl`](https://diffeq.sciml.ai/stable/features/callback_functions/)
```

**Comment by kpamnany** [comment]
Please keep line widths under 80?

**Comment by kmdeck** [comment]
> Please keep line widths under 80?

will do! thanks!

File: docs/src/GettingStarted/Terminology.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Terminology
+
+The `ClimateMachine` documentation uses terminology from several disciplines. Below are some definitions of some of these terms. Please let us know via a comment if there are others that you find confusing, and we will add them.
+
+## Software terms
+
+* Driver file: a julia script which (a) specifies which [`BalanceLaw`](#ClimateMachine.BalanceLaws.BalanceLaw) is being used, (b) specifies changes to the default version of that model, (c) specifies initial and boundary conditions, (d) specifies numerical details related to timestep, grid, and frequency of output, and (e) runs the integration. Examples can be found in tutorials and experiments.
```
Comment (kmdeck) [inline docs/src/GettingStarted/Terminology.md:7]: these types of links dont work. TBD
also would be nice to bold the terms being defined.

Comment (charleskawczynski) [reply docs/src/GettingStarted/Terminology.md:7]: I think you need [`BalanceLaw`](@ref ClimateMachine.BalanceLaws.BalanceLaw)

Comment (kmdeck) [reply docs/src/GettingStarted/Terminology.md]: this ended up working...
[`BalanceLaw`](../APIs/BalanceLaws/BalanceLaws.md#ClimateMachine.BalanceLaws.BalanceLaw)

@ref gave an error? not sure, that works in tutorials!

Comment (kmdeck) [reply docs/src/GettingStarted/Terminology.md]: > this ended up working...
> [`BalanceLaw`](../APIs/BalanceLaws/BalanceLaws.md#ClimateMachine.BalanceLaws.BalanceLaw)
> 
> @ref gave an error? not sure, that works in tutorials!

File: docs/src/GettingStarted/Terminology.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+`ClimateMachine.jl` uses
+[`KernelAbstractions.jl`](https://github.com/JuliaGPU/KernelAbstractions.jl)
+as its kernel language.
+* __Callback__: Functions executed by the ODE integrator after each time step; see
+[`solve!`](#ClimateMachine.ODESolvers.solve!).
+This allows the ability to inject custom behavior into the ODE
+integrators, such as diagnostic output, visualization, and apply
+filtering to the numerical solution.
+Several convenience functions exists specifying
+[callback](../APIs/Numerics/ODESolvers/ODESolvers.md/#ClimateMachine.ODESolvers.GenericCallback)
```
Comment (kmdeck) [inline docs/src/GettingStarted/Terminology.md:33]: Not sure why this link is not working. It copies what does work, for the BalanceLaw API.

**Comment by kmdeck** [comment]
Im having trouble with the formatting in the Terminology.md page. The line breaks are strange - sometimes double spaced, sometimes not. I cant reproduce the error in VS markdown preview. If anyone has any ideas, let me know :)

File: docs/src/GettingStarted/Terminology.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
+# Terminology
+
+The `ClimateMachine` documentation uses terminology from several disciplines. Below are some definitions of some of these terms. Please let us know via a comment if there are others that you find confusing, and we will add them.
```
Comment (jiahe23) [inline docs/src/GettingStarted/Terminology.md:3]: I have an impression that we are not adding a comment session for this release. Shall we change it to "via email/issue"?

Comment (kmdeck) [reply docs/src/GettingStarted/Terminology.md:3]: good point! Ill fix this.

**Review by jiahe23 — APPROVED** [review]

**Pull request merged**